syntax = "proto2";

package xxua;

enum ProtoTestEnum {
  ProtoTestEnumOne = 1;
  ProtoTestEnumTwo = 2;
  ProtoTestEnumThree = 3;
};

message ProtoTestSubmessage {
  optional int64 i = 1;
  optional string s = 2;
}

message ProtoTestMessage {
  optional double optional_double_field = 1;
  optional float optional_float_field = 2;
  optional int32 optional_int32_field = 3;
  optional int64 optional_int64_field = 4;
  optional uint32 optional_uint32_field = 5;
  optional uint64 optional_uint64_field = 6;
  optional sint32 optional_sint32_field = 7;
  optional sint64 optional_sint64_field = 8;
  optional fixed32 optional_fixed32_field = 9;
  optional fixed64 optional_fixed64_field = 10;
  optional sfixed32 optional_sfixed32_field = 11;
  optional sfixed64 optional_sfixed64_field = 12;
  optional bool optional_bool_field = 13;
  optional string optional_string_field = 14;
  optional bytes optional_bytes_field = 15;
  optional ProtoTestEnum optional_enum_field = 16;
  optional ProtoTestSubmessage optional_message_field = 17;

  repeated double repeated_double_field = 18;
  repeated float repeated_float_field = 19;
  repeated int32 repeated_int32_field = 20;
  repeated int64 repeated_int64_field = 21;
  repeated uint32 repeated_uint32_field = 22;
  repeated uint64 repeated_uint64_field = 23;
  repeated sint32 repeated_sint32_field = 24;
  repeated sint64 repeated_sint64_field = 25;
  repeated fixed32 repeated_fixed32_field = 26;
  repeated fixed64 repeated_fixed64_field = 27;
  repeated sfixed32 repeated_sfixed32_field = 28;
  repeated sfixed64 repeated_sfixed64_field = 29;
  repeated bool repeated_bool_field = 30;
  repeated string repeated_string_field = 31;
  repeated bytes repeated_bytes_field = 32;
  repeated ProtoTestEnum repeated_enum_field = 33;
  repeated ProtoTestSubmessage repeated_message_field = 34;
};

